templates/python/produceSkeletons_D41_NoSmear_PU_AVE_200_BX_25ns_aged.sh

```diff
@@ -0,0 +1,122 @@
+#!/bin/sh
+
+# This is the shell script that will generate all the skeletons using cmsDriver commands.
+# The commands included have been taken from runTheMatrix with the following command:
+#
+# runTheMatrix.py -w upgrade -l 20634.0 --command="--no_exec" --dryRun
+#
+# The reconstruction as part of the ticl framework is injected into the RECO_fragment.
+#
+# For all commands remove --filein and --fileout options.
+# Add python_filename option
+#
+# The first command combines step1 and step2 (GSD):
+# - mix in pileup
+# - run up to DIGI...HLT:@fake2
+# The following changes are implemented on top:
+# --beamspot HLLHC14TeV ➜ --beamspot NoSmear
+# --eventcontent FEVTDEBUG ➜ --eventcontent FEVTDEBUGHLT
+# Removed --relval option.
+#
+# The second command is step3 removing overlap with step2 (RECO):
+# - remove pileup part
+# - remove MINIAODSIM from event content and data tier
+# - remove PAT from steps (-s)
+# - remove @miniAODValidation from VALIDATION step
+# - remove @miniAODDQM from DQM step
+#
+# The third command is a copy of the second only re-running RECO (for NTUP):
+# - remove DQM from event content
+# - remove DQMIO from data tier
+# - add --processName=NTUP option
+#
+# Those commands should be regularly checked and, in case of changes, propagated into this script!
+
+action() {
+  # default arguments
+  local inject_ticl="1"
+  local geometry="Extended2026D41"
+  local pileup_input="das:/RelValMinBias_14TeV/CMSSW_10_6_0_patch2-106X_upgrade2023_realistic_v3_2023D41noPU-v1/GEN-SIM"
+  local custom=""
+
+  # parse arguments
+  for arg in "$@"; do
+    if [ "$arg" = "ticl" ]; then
+      inject_ticl="1"
+    elif [ "$arg" = "no-ticl" ]; then
+      inject_ticl="0"
+    elif [[ $arg =~ "geometry" ]]; then        
```

Comment:
Hi!
The regexp operator `=~` will also evaluate to True when "geometry" is located at an arbitrary position in the argument, e.g. when the pileup file string contains "geometry" somewhere in the campaign description. I would explicitly check for the beginning of the string and go with
```bash
[[ "$arg" =~ ^geometry= ]]
```
to be safe (same for the other args below).

Reply:
ok thanks, i added ^ to all arguments